src/index.ts

```diff
@@ -36,6 +36,46 @@ async function redirectWithFlash(url: string, flash: FlashSessionValues, init?:
   });
 }
 
+/**
+ * Helper method used to display a success toast notification without redirection
+ *
+ * It need to be awaited
+ * @param passedInData Generic object containing the data
+ * @param toast Message to be shown as success
+ * @param init Additional response options (status code, additional headers etc)
+ * @returns Returns data and a toast cookie set
+ */
+export async function jsonWithSuccess<T>(passedInData: T, toast: string, init?: ResponseInit) {
```

Comment:
sorry for the delay didn't have time to review again, so I would do the following, I'd create an utility called `jsonWithFlash` that is not exported that has the following signature:
```ts
async function jsonWithFlash<T>(data: T, flash: FlashSessionValues, init?: ResponseInit) {
  return json(data, {
    ...init,
    headers: await flashMessage(flash, init?.headers),
  });
} 
```
Then after that create a `jsonWithToast` like so:
```ts
export function jsonWithToast<T>(data: T, toast: ToastMessage, init?: ResponseInit) {
  return jsonWithFlash(data, { toast }, init);
}
```
And from there have these utilities you have created, eg:

```ts
export function jsonWithSuccess<T>(data: T, message: string, init?: ResponseInit) {
  return jsonWithToast(data, { message, type: "success" }, init);
}
```
Just so it's consitent with the redirect methods. Also I'd add jsonWithWarning and jsonWithInfo variants. If you want I can add that on top, or would you like to? @Erij-Maherzia-BEN-BRAHIM 

Also, thank you so much for this contribution :)

Reply:
Hi @AlemTuzlak, Thank you for your feedback and suggestions! I appreciate your time in reviewing the code. I will certainly implement the changes you've recommended to ensure consistency with the redirect methods.